uvicorn/importer.py

```diff
@@ -22,17 +22,19 @@ def import_from_string(import_str: Any) -> Any:
     except ImportError as exc:
         if exc.name != module_str:
             raise exc from None
```

Comment:
Do you know, or would you like to investigate why this conditional exists? There is no test that covers this... 🤔

Reply:
If the string-imported module imports a module which does not exist the exc.name and the module_str will have different values - since the exception is being raised on the import. This is now caught by `ModuleImportError` and reported correctly so i have taken this check out.